### PR TITLE
adapt tests for dual-stack cluster: [test_id:4153]VMI with masquerade binding and guest agent should expose Pod IP as its public address

### PR DESCRIPTION
**Pull request by ormergi**
**What this PR does / why we need it**:
This PR adapt the test `[test_id:4153]VMI with masquerade binding and guest agent should expose Pod IP as its public address` 
for dual-stack clusters
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Rebased on top #3958 (1 commit)

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by oshoval** [comment]
Add please hold until #3958 is merged as we done on the others.

File: tests/vmi_configuration_test.go
Comment (maiqueb) [inline tests/vmi_configuration_test.go:2521]: I've asked @oshoval this before; could you guys coordinate and provide a single function to make sure the IPs are consistent between the pod & the vmi ? 

pretty clear everyone will need it.

File: tests/vmi_configuration_test.go
Comment (oshoval) [inline tests/vmi_configuration_test.go:2525]: you iterate according `vmiVirtLauncherPod.Status.PodIP` len
and accessing `vmi.Status.Interfaces[0].IPs`
in case len of `vmi.Status.Interfaces[0].IPs` < len of `vmiVirtLauncherPod.Status.PodIP` there would be a panic,
i would suggest to add len check verify please

**Review by oshoval — CHANGES_REQUESTED** [review]

File: tests/vmi_configuration_test.go
Comment (maiqueb) [inline tests/vmi_configuration_test.go:2524]: I think you should assert the length of the IPs between pod / VMI.

I also personally prefer to iterate the IPs rather than use indexes; I'd use  `for _, ip := range vmiVirtLauncherPod.Status.PodIPs { ... }` instead, but up to you.

**Comment by oshoval** [comment]
Please add hold until #3958 is merged, as we did on the others.

**Review by maiqueb — CHANGES_REQUESTED** [review]
Looks good.

**Comment by ormergi** [comment]
/hold
Until #3958 is merged

File: tests/vmi_configuration_test.go
Comment (ormergi) [reply tests/vmi_configuration_test.go:2521]: Which package should contain it?

File: tests/vmi_configuration_test.go
Comment (ormergi) [reply tests/vmi_configuration_test.go:2524]: I iterate by an index to check the order of the addresses as well, I mention it in the commit message.

File: tests/vmi_configuration_test.go
Comment (ormergi) [reply tests/vmi_configuration_test.go:2525]: You right, I will add a check for length of both arrays.

I intentionally dont use `range` because I am not sure there is grantee that it will return ascending sequence of `0,1..n`

Comment (oshoval) [reply tests/vmi_configuration_test.go:2525]: on maps it wont reserve order, on array / slice it does
and here even if it doesnt you dont need order

**Comment by ormergi** [comment]
Changed `Consistently` function to return error instead of `bool`
Add arrays length checks

File: tests/vmi_configuration_test.go
Comment (oshoval) [reply tests/vmi_configuration_test.go]: Done on my PR, extracted to external file.

Comment (ormergi) [reply tests/vmi_configuration_test.go]: This is not merged code I can't use it

Comment (oshoval) [reply tests/vmi_configuration_test.go]: you can cherry-pick the patch , and put hold, there is just one commit that have the function that you need

Comment (ormergi) [reply tests/vmi_configuration_test.go]: At the current state it won't work for this test for the reason I mentioned 
If this PR merge first just refactor this test
I see no reason to hold this PR

Comment (oshoval) [reply tests/vmi_configuration_test.go]: i dont think i agree on that reason, also Miguel said it to you i think in a thread below.
would be best that you express the reason itself please that you think, anyhow do what you select,
i will continue mine.

Comment (oshoval) [reply tests/vmi_configuration_test.go]: https://github.com/kubevirt/kubevirt/pull/4042#discussion_r478612964
I tried to answer why there isnt an issue, lets discuss it until we agree please, imo this what we should do.

**Comment by ormergi** [comment]
@phoracek @qinqon @EdDev  CI is green

File: tests/vmi_configuration_test.go
Comment (phoracek) [inline tests/vmi_configuration_test.go:2522]: don't

File: tests/vmi_configuration_test.go
Comment (phoracek) [inline tests/vmi_configuration_test.go:2527]: lengths don't

File: tests/vmi_configuration_test.go
Comment (phoracek) [inline tests/vmi_configuration_test.go:2537]: No need for 1* if it is just one second

**Review by phoracek — COMMENTED** [review]
Minor nits in the error messages. I'm fine with this going in as is though.

/approve

Let's get one lgtm from people previously involved in the review

File: tests/vmi_configuration_test.go
Comment (RamLavi) [inline tests/vmi_configuration_test.go:2521]: looks like you're doing some very similar to `ValidateVMIandPodIPMatch` in [another PR](https://github.com/kubevirt/kubevirt/pull/4042) - can you maybe use that? no need for both implementations..

**Review by RamLavi — CHANGES_REQUESTED** [review]
Maybe it's worth to use `ValidateVMIandPodIPMatch` from @oshoval 's [PR](https://github.com/kubevirt/kubevirt/pull/4042) (once it's merged of course)

**Comment by oshoval** [comment]
There are some nits as Petr said, and few more imo (some already were written before), but its fine to continue as its now.
/lgtm

**Pull request merged**
